src/garage/tf/policies/gaussian_mlp_task_embedding_policy.py

```diff
@@ -0,0 +1,283 @@
+            std_hidden_nonlinearity=std_hidden_nonlinearity,
+            std_output_nonlinearity=std_output_nonlinearity,
+            std_parameterization=std_parameterization,
+            layer_normalization=layer_normalization,
+            name='GaussianMLPModel')
+
+        self._initialize()
+
+    def _initialize(self):
+        obs_input = tf.compat.v1.placeholder(tf.float32,
+                                             shape=(None, self._obs_dim))
+        # task_input = self._encoder.input
+        latent_input = tf.compat.v1.placeholder(
+            tf.float32, shape=(None, self._encoder.output_dim))
+
+        with tf.compat.v1.variable_scope(self.name) as vs:
+            self._variable_scope = vs
+
+            with tf.compat.v1.variable_scope('concat_obs_latent'):
+                obs_latent_input = tf.concat([obs_input, latent_input],
+                                             axis=-1)
+            self.model.build(obs_latent_input, name='from_latent')
+
+        self._f_dist_obs_latent = tf.compat.v1.get_default_session(
+        ).make_callable([
+            self.model.networks['from_latent'].mean,
+            self.model.networks['from_latent'].log_std
+        ],
+                        feed_list=[obs_input, latent_input])
+
+    def dist_info_sym_under_task(self,
+                                 obs_var,
+                                 task_var,
+                                 state_info_vars=None,
+                                 name='default'):
+        """Build a symbolic graph of the action distribution given task.
+
+        Args:
+            obs_var (tf.Tensor): Symbolic observation input.
+            task_var (tf.Tensor): Symbolic task input.
+            state_info_vars (dict): Extra state information, e.g.
+                previous action.
+            name (str): Name for symbolic graph.
+
+        Returns:
+            dict[tf.Tensor]: Outputs of the symbolic graph of
+                action distribution parameters.
+
+        """
+        with tf.compat.v1.variable_scope(self._variable_scope):
+            latent_dist_info_sym = self._encoder.dist_info_sym(task_var,
+                                                               name=name)
+            latent_var = self._encoder.distribution.sample_sym(
+                latent_dist_info_sym)
+            obs_latent_input = tf.concat([obs_var, latent_var], axis=-1)
+            mean_var, log_std_var, _, _ = self.model.build(obs_latent_input,
+                                                           name=name)
+        return dict(mean=mean_var, log_std=log_std_var)
+
+    def dist_info_sym_under_latent(self,
+                                   obs_var,
+                                   latent_var,
+                                   state_info_vars=None,
+                                   name='from_latent'):
+        """Build a symbolic graph of the action distribution given latent.
+
+        Args:
```

Comment:
please provide shape documentation, and tell us what's in those dicts

Reply:
`None` is not a valid shape in shape documentation. It seems you are referring to a batch size of some sort, so tell your user how to determine what that batch size will be. 

see https://github.com/rlworkgroup/garage/blob/master/src/garage/_dtypes.py for examples

Reply:
Z would be a more typical symbol for a latent, by the way.